src/components/editor/ArticleListController.jsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import { browserHistory } from 'react-router';
+import FalcorController from "lib/falcor/FalcorController";
+import List from "./List";
+
+var pageLength = 50;
+
+export default class ArticleListController extends FalcorController {
+  constructor(props) {
+    super(props);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleUpdateSlugValue = this.handleUpdateSlugValue.bind(this);
+    this.safeSetState({
+      slugSearchValue: ""
+    })
+  }
+  static getFalcorPath(params) {
+    return [['articlesByTime', 'paginated', {from: pageLength*(params.page-1), length: pageLength}, ['title', 'slug']], ['articlesByTime', 'paginated', 'total']];
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    let pathStart = "/articles/page/" + this.props.params.page + "/article/slug/";
+    browserHistory.push(pathStart + this.state.slugSearchValue + "/");
+    this.safeSetState({
+      slugSearchValue: ""
+    });
+  }
+
+  handleUpdateSlugValue(event) {
+    this.safeSetState({
+      slugSearchValue: event.target.value
+    });
+  }
+
+  handlePageChange(increment, event) {
+    let path = this.props.location.pathname;
+    path = path.split('/');
+    path[3] = (parseInt(path[3])+increment).toString();
+    browserHistory.push(path.join('/'));
+  }
+
+  render() {
+      // For Ling when code reviewing. I was choosing between two style decisions here when making the loading look as I wanted it
+      // I could either do as I did now with quite a few ternary operators. Or I could've put it all in a big if this.state.ready
+      // So that it only processed the data if it had it. But in that case I then just had to put in quite a few var's instead of let's
+      // So they would carry over to the return statement. Which would you think is best?
+      let data = this.state.data ? this.state.data.articlesByTime.paginated : null;
```

Comment:
Well the point is that at the moment the Falcor Router get function is relying on the fact that you can only make paginated requests, aka. it can only be a single range, and start and end have to be able to fit into a single page that has started at index 0 (mathematically I guess something like end = start+gcd(start, end)).

If we don't have these restrictions and the Ghost API is paginated in its structure, I would either have to query waaay too much data to make sure I got it all in one fetch, or do two queries to the Ghost API for every fetch , or more complicated stuff depending on how many restrictions there are on it.
I had begun doing the general one, but realized probably no one will need it so it's a waste of time writing it. Kept the `['paginated']` part though just to make it clear (and there's a comment in the Falcor Router explaining it) that you could only use this for paginated results.